### PR TITLE
Interview GUI

### LINKED ISSUE #91: GUI for Interviews
GUI for Interviews

### LINKED ISSUE #107: Split GUI screen
... into List of IA on the left and details of IA on the right.

### PR DESCRIPTION
**Pull request by foochifa**
Added:
- InternshipApplicationDetail panel
- InterviewList 
- onclick selection to show IA details
- Select Command
- Display IA from add/edit/interview

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/CommandResult.java [hunk 1]
```diff
@@ -38,6 +48,19 @@ public CommandResult(String feedbackToUser) {
...
+     * Constructs a {@code CommandResult} with the specified {@code feedbackToUser},
+     * also instructs window to display an internship application at specified {@code internshipApplicationIndex}.
+     */
+    public CommandResult(String feedbackToUser, InternshipApplication internshipApplicationToShow) {
+        this.feedbackToUser = requireNonNull(feedbackToUser);
+        this.showInternshipApplication = true;
+        this.internshipApplicationToShow = requireNonNull(internshipApplicationToShow);
+        showHelp = false;
+        showStatistics = false;
+        exit = false;
```
Comment (gerhean) [inline src/main/java/seedu/address/logic/commands/CommandResult.java:61]: Why does CommandResult need to know the internshipApplication to show?
Why not just set it in the model?

Comment (foochifa) [reply src/main/java/seedu/address/logic/commands/CommandResult.java:61]: So two reasons why I ended up choosing this implementation:

1. The Display handling can only be done in MainWindow and it only has a direct association with Logic. Hence, I think it makes it more streamlined if i were to handle it in MainWindow's execution method.
2. At first I wanted to use Index and acquire the IA from getFilteredList in Logic. However, I realised this isn't feasible with add command. As you can add an internship application that is filtered out by the list, which gives an IndexOutOfBoundsException

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -140,6 +140,15 @@ public static void assertCommandSuccess(Command command, Model actualModel, Stri
         assertCommandSuccess(command, actualModel, expectedCommandResult, expectedModel);
     }
 
+    /**
+     * Compares to a command that takes in an internship application to display
+     */
+    public static void assertCommandSuccess(Command command, Model actualModel, String expectedMessage,
+                                            Model expectedModel, InternshipApplication internshipApplication) {
```
Comment (ongch00) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: The styling indentation here is a bit off

File: src/main/java/seedu/address/ui/InternshipApplicationDetail.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        email.setText(
+                "Email:                      " + internshipApplication.getEmail().value);
+        role.setText(
+                "Role:                        " + internshipApplication.getRole().fullRole);
+        priority.setText(
+                "Priority:                   " + internshipApplication.getPriority().fullPriority);
+        applicationDate.setText(
+                "Application Date:    " + internshipApplication.getApplicationDate().printDate());
+        status.setText(
+                "Status:                     " + internshipApplication.getStatus().toString());
```
Comment (wxwxwxwx9) [inline src/main/java/seedu/address/ui/InternshipApplicationDetail.java]: I think the spacing can be abstracted out into a private variable "whitespace"?

Would be good in terms of reusability as well.

**Review by wxwxwxwx9 — COMMENTED** [review]
Overall looks okay!

**Review by ongch00 — COMMENTED** [review]
Other than that one line, there is no other obvious issues.

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 2 above)
Comment (foochifa) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: Okay will fix this thanks for the catch

File: src/main/java/seedu/address/ui/InternshipApplicationDetail.java (hunk 3 above)
Comment (ongch00) [reply src/main/java/seedu/address/ui/InternshipApplicationDetail.java]: I think would be better to have private constants for the whole thing like `private static final String PHONE = "Phone:                     "` as the white space is not of equal length

Comment (foochifa) [reply src/main/java/seedu/address/ui/InternshipApplicationDetail.java]: Yeah good idea thanks!

Comment (gerhean) [reply src/main/java/seedu/address/ui/InternshipApplicationDetail.java]: https://stackoverflow.com/questions/388461/how-can-i-pad-a-string-in-java/391978#391978

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 2 above)
Comment (gerhean) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: Please use the auto formatter provided by intellij

**Comment by wxwxwxwx9** [comment]
Looks good!

**Pull request merged**